python/python/bystro/prs/preprocess_for_prs.py

```diff
@@ -184,39 +195,41 @@ def generate_thresholded_overlap_scores_dosage(
     gwas_scores_path: str, dosage_matrix_path: str, p_value_threshold: float
 ) -> pd.DataFrame:
     """Compare and restrict to overlapping loci between dosage matrix and thresholded scores."""
-    scores = _preprocess_scores(gwas_scores_path)
-    dosage_loci_nomiss = _extract_nomiss_dosage_loci(dosage_matrix_path)
+    thresholded_scores_chunks = []
 
-    thresholded_indices_set = set()
-    for chunk in read_feather_in_chunks(gwas_scores_path, columns=["P"], chunk_size=1000):
+    for chunk in read_feather_in_chunks(gwas_scores_path, chunk_size=1000):
         thresholded_indices = get_p_value_thresholded_indices(chunk, p_value_threshold)
-        thresholded_indices_set.update(thresholded_indices)
-    scores_filtered_indices = sorted(list(thresholded_indices_set))
-    thresholded_scores = scores.iloc[scores_filtered_indices]
-    thresholded_scores = thresholded_scores.set_index("SNPID")
+        if thresholded_indices:
+            thresholded_scores_chunks.append(chunk.loc[thresholded_indices])
 
-    if len(thresholded_scores) == 0:
+    if not thresholded_scores_chunks:
```

Comment:
ValueErrors are reserved for cases where the input arguments are wrong. Lets use RuntimeError here